File: fastmq-common/src/main/java/com/song/fastmq/storage/common/concurrent/SafeRunnable.java
package com.song.fastmq.storage.common.concurrent;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by song on 2017/11/5.
 */
public abstract class SafeRunnable implements Runnable {

    private static final Logger logger = LoggerFactory.getLogger(SafeRunnable.class);

    @Override public void run() {
        try {
            safeRun();
        } catch (Throwable e) {
            logger.error("Unknown exception", e);
        }
    }

    public abstract void safeRun();
}

File: fastmq-common/src/main/java/com/song/fastmq/common/concurrent/SafeRunnable.kt
package com.song.fastmq.common.concurrent

import org.slf4j.LoggerFactory

/**
 * Created by song on 2017/11/5.
 */
abstract class SafeRunnable : Runnable {

    override fun run() {
        try {
            safeRun()
        } catch (e: Throwable) {
            logger.error("Uncaught exception", e)
        }
    }

    abstract fun safeRun()

    companion object {

        private val logger = LoggerFactory.getLogger(SafeRunnable::class.java)
    }
}


